sympy/core/basic.py

```diff
@@ -319,6 +319,16 @@ def inner_key(arg):
         args = len(args), tuple([inner_key(arg) for arg in args])
         return self.class_key(), args, S.One.sort_key(), S.One
 
+    def _do_eq_sympify(self, other):
+        for superclass in type(other).__mro__:
+            try:
+                return self == _external_converter[superclass](other)
+            except KeyError:
+                continue
```

Comment:
This `try/except` pattern (which already exists in the codebase) should not be used as it can also catch unintended `KeyError` e.g. from calling the converter function or from the equality comparison `==` itself. It should be structured more like:
```python
for superclass in type(other).__mro__:
    try:
        return conv = _external_converter[superclass]
    except KeyError:
        continue
    else:
        return self == conv(other)
```
Personally I would just avoid the `try/except` altogether and do:
```python
for superclass in type(other).__mro__:
    conv = _external_converter.get(superclass)
    if conv is not None:
        return self == conv(other)
```